### PR TITLE
Update mtproto_sender.py

### PR DESCRIPTION
**Pull request by maxim-smirnov**
#453

### CODE REVIEW

**Comment by Lonami** [comment]
Have you deleted the "broken" session file? Does this happen with the check that we added? It would be better if we figured out why the padding is wrong.

**Comment by maxim-smirnov** [comment]
I don't store the session files. Even I'm not authing with any phone number. I'm using API without authentication.

**Comment by maxim-smirnov** [comment]
To catch exception definitely better than get error from C lib)

File: telethon/network/mtproto_sender.py [hunk 1]
```diff
@@ -158,8 +158,11 @@ def _decode_msg(self, body):
             msg_key = reader.read(16)
 
             key, iv = utils.calc_key(self.session.auth_key.key, msg_key, False)
+            data = reader.read(len(body) - reader.tell_position())
+            if len(data) % 16 != 0:
+                raise SecurityError('AES block size missmatch')
```
Comment (Lonami) [inline telethon/network/mtproto_sender.py]: You have misspelt "missmatch" twice now. It's "mismatch".

File: telethon/network/mtproto_sender.py [hunk 2]
```diff
@@ -158,8 +158,11 @@ def _decode_msg(self, body):
...
             plain_text = AES.decrypt_ige(
-                reader.read(len(body) - reader.tell_position()), key, iv)
+                data, key, iv)
```
Comment (Lonami) [inline telethon/network/mtproto_sender.py]: No point in writing this in a new line as it's short enough to fit in one:
```python
            plain_text = AES.decrypt_ige(data, key, iv)
```

File: telethon/network/mtproto_sender.py [hunk 3]
```diff
@@ -158,8 +158,11 @@ def _decode_msg(self, body):
             msg_key = reader.read(16)
 
             key, iv = utils.calc_key(self.session.auth_key.key, msg_key, False)
+            data = reader.read(len(body) - reader.tell_position())
+            if len(data) % 16 != 0:
```
Comment (Lonami) [inline telethon/network/mtproto_sender.py]: I'm still not sure about this… How can it happen? I would rather we found the root of the problem.

**Pull request closed without merging**